### PR TITLE
[BugFix] Fix potential message lost in routine load

### PR DESCRIPTION
**Pull request by chen9t**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

No issues proposed for now.

## Problem Summary(Required) ：

The return value of `KafkaConsumerPipe::append_data` was not processed, message could get lost if we failed to append.

### CODE REVIEW

File: be/src/runtime/routine_load/data_consumer_group.cpp [hunk 1]
```diff
@@ -171,8 +170,8 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
             VLOG(3) << "get kafka message"
                     << ", partition: " << msg->partition() << ", offset: " << msg->offset() << ", len: " << msg->len();
 
-            (kafka_pipe.get()->*append_data)(static_cast<const char*>(msg->payload()), static_cast<size_t>(msg->len()),
-                                             row_delimiter);
+            Status st = (kafka_pipe.get()->*append_data)(static_cast<const char*>(msg->payload()),
```
Comment (rickif) [inline be/src/runtime/routine_load/data_consumer_group.cpp:173]: Maybe it's better to add log of the `st` detail at line 183.

**Comment by rickif** [comment]
> [CLA assistant check](https://cla-assistant.io/StarRocks/starrocks?pullRequest=6586) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=6586) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=6586) it.

@chen9t
Thanks for your contribution! And could you please sign the CLA ?

File: be/src/runtime/routine_load/data_consumer_group.cpp (hunk 1 above)
Comment (chen9t) [reply be/src/runtime/routine_load/data_consumer_group.cpp:173]: Error message added.

**Comment by chen9t** [comment]
> > [CLA assistant check](https://cla-assistant.io/StarRocks/starrocks?pullRequest=6586) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=6586) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=6586) it.
> 
> @chen9t Thanks for your contribution! And could you please sign the CLA ?

CLA signed, thank you!

File: be/src/runtime/routine_load/data_consumer_group.cpp [hunk 2]
```diff
@@ -181,7 +180,7 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
                 VLOG(3) << "consume partition[" << msg->partition() << " - " << msg->offset() << "]";
             } else {
                 // failed to append this msg, we must stop
-                LOG(WARNING) << "failed to append msg to pipe. grp: " << _grp_id;
+                LOG(WARNING) << "failed to append msg to pipe. grp: " << _grp_id << ", errmsg=" << st.get_error_msg();
```
Comment (rickif) [inline be/src/runtime/routine_load/data_consumer_group.cpp:183]: LOG(WARNING) << "failed to append msg to pipe. grp: " << _grp_id << ", errmsg=" << st;

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_clang_format

**Review by chaoyli — APPROVED** [review]

**Pull request merged**